pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -619,7 +635,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			InitContainers:     initContainers,
 			Containers: []v1.Container{
 				{
-					Command:         osdStartScript(command, args, c.spec.Storage.FlappingRestartIntervalHours),
+					Command:         append([]string{"bash", "-x", "-c", cephOSDStart}, command...),
```

Comment:
Thanks for the example of keeping the arguments consistent whether we decide to use the script or not. So supporting both ways of starting the OSD would just still require setting a different command, to either use `bash` or not. 

To support both, a different liveness probe would be needed (one checks for the touched temp file, and one doesn't). But we need to support both liveness probes anyway for other daemons. 

Looking back at #12715, maybe it's not that complicated to support both ways of launching the OSD. @sp98 Can you clarify the complication of looking of the existing properties for the osd info? I think that's not a concern, we could just generate the needed `ceph-osd` or `bash` command in the OSD pod spec depending on the cephcluster CR or operator env var settings. We should need to look at the existing OSD pod to figure out how to generate the spec for this command.

Reply:
I tried out this commit as is. If failed to start the OSD. 

```
❯ oc logs rook-ceph-osd-0-54ffb568f-h97pm -n rook-ceph
Defaulted container "osd" out of: osd, log-collector, activate (init), expand-bluefs (init), chown-container-data-dir (init)
+ set -o nounset
+ RESTART_INTERVAL=6
+ child_pid=
+ sigterm_received=false
+ trap sigterm SIGTERM
+ child_pid=530
+ wait 530
+ --foreground --id 0 --fsid 4806bee1-9a4b-42bf-9c31-88f9b77302f3 --setuser ceph --setgroup ceph '--crush-location=root=default host=minikube' --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true '--default-log-stderr-prefix=debug ' --default-log-to-file=false --default-mon-cluster-log-to-file=false --ms-learn-addr-from-peer=false
ceph-osd: line 15: --foreground: command not found
+ wait 530
+ ceph_osd_rc=127
+ '[' 127 -eq 0 ']'
+ exit 127
```

Reply:
> To support both, a different liveness probe would be needed (one checks for the touched temp file, and one doesn't). But we need to support both liveness probes anyway for other daemons.

I think the liveness probe does its check unless the file is present right? Doesn't that behavior also fit for the method of running osds that doesn't use the script?

Reply:
@travisn I tried testing the changes that Blaine has suggested in this PR. Blaine's approach of writing updating the OSD scripts would work and make it easier to fetch the OSDInfo from the POD.  So we can use both the approaches (osd script and directly running osd cmd) in Rook without make a lot of changes. 

We need not have a different liveless probe to support both the scenarios. Liveness probe can always check for the touched temp file whether osd is running as a bash script or just the osd command.